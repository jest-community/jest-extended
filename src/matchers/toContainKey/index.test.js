import matcher from './';

expect.extend(matcher);

const data = { hello: 'world' };

describe('.toContainKey', () => {
  it('passes when given object contains key', () => {
    expect(data).toContainKey('hello');
  });

  it('fails when given object does not contain key', () => {
    expect(() => expect(data).toContainKey('missing')).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toContainKey', () => {
  it('passes when given object does not contain key', () => {
    expect(data).not.toContainKey('missing');
  });

  it('fails when given object contains key', () => {
    expect(() => expect(data).not.toContainKey('hello')).toThrowErrorMatchingSnapshot();
  });
});
