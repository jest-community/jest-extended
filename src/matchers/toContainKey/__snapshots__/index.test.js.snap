// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toContainKey fails when given object contains key 1`] = `
"expect(received).not.toContainKey(expected)

Expected object to not contain key:
  \\"hello\\"
Received:
  {\\"hello\\": \\"world\\"}"
`;

exports[`.toContainKey fails when given object does not contain key 1`] = `
"expect(received).toContainKey(expected)

Expected object to contain key:
  \\"missing\\"
Received:
  {\\"hello\\": \\"world\\"}"
`;
